File: Project2-backend/Project2-frontend/src/app/services/ticket.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class TicketService {

  private baseUrl = 'http://localhost:8080/tickets';

  constructor(private http: HttpClient) { }

  addTicket(ticket: Object):Observable<Object>{
    return this.http.post(`${this.baseUrl}`, ticket)
  }

  getAllTickets(): Observable<any> {
    return this.http.get(`${this.baseUrl}`);
  }

  getTicket(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  deleteTicket(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`, {responseType: 'text'});
  }

  updateTicket(id: number, value:any): Observable<Object>{
    return this.http.put(`${this.baseUrl}/${id}`, value);
  }
}

File: Project2-backend/src/main/java/com/revature/services/TicketService.java
package com.revature.services;

import java.util.List; 

import com.revature.models.Ticket;

public interface TicketService {
	
	public Ticket addTicket(Ticket ticket);
	public List<Ticket> getAllTickets();
	public Ticket getTicket(int id);
    public Ticket updateTicket(Ticket ticket);
    public boolean deleteTicket(int id);
}
